Gems/GradientSignal/Code/Tests/GradientSignalImageTests.cpp

```diff
@@ -464,6 +466,42 @@ namespace UnitTest
         RunPixelTest(test);
     }
 
+    TEST_F(GradientSignalImageTestsFixture, ImageGradientComponentAdvancedBilinearSampling)
+    {
+        // Set one pixel, map Gradient 1:1 to lookup space, get same pixel back
+        PixelTestSetup test =
+        {
+            4, AZ::Vector2(0, 0), {200, 150, 100, 50},                  // Source image:  4 x 4 with (0, 0) set to different values in each channel
+            4, 1.0f,                                                    // Mapped Shape:  4 x 4 with tiling (1.0, 1.0), unbounded
+            GradientSignal::WrappingType::None,
+            4, 1.0f, 1.0f,                                              // Validate that in 4 x 4 range, only 0, 0 is set
+            { AZ::Vector2(0, 0), PixelTestSetup::EndOfList },
+            true,                                                       // Enabled the advanced mode
+            GradientSignal::ChannelToUse::Red,                          // Use Red channel (default)
+            GradientSignal::CustomScaleType::None,                      // No custom scale (default)
+            0.0f,                                                       // Min scale (won't be used since None scale type is set)
+            1.0f,                                                       // Max scale (won't be used since None scale type is set)
+            GradientSignal::SamplingType::Bilinear                      // Enable the bilinear sampling type
+        };
+
+        // Compute the delta X/Y for sampling pixel
+        float deltaX = test.m_pixel.GetX() - floor(test.m_pixel.GetX());
+        float deltaY = test.m_pixel.GetY() - floor(test.m_pixel.GetY());
+
```

Comment:
Is this test really testing bilinear sampling? It seems like deltaX and deltaY will be 0, so we'll just get back the pixel value we set. If that's the case, it would be better to have a test that tests a query at non-zero deltaX and deltaY. It might require a more custom test setup than what RunPixelTest() supports though. Or you could change PixelTestSetup, though it's admittedly already grown a bit out-of-hand for readability...

Reply:
Good point, I have updated the unit test to perform a different pixel test for this case